dali/pipeline/executor/executor.h

```diff
@@ -142,6 +147,16 @@ class DLL_PUBLIC Executor : public ExecutorBase, public QueuePolicy {
 
   DISABLE_COPY_MOVE_ASSIGN(Executor);
 
+  /**
+   * Returns the current checkpoint.
+  */
+  DLL_PUBLIC Checkpoint& GetCurrentCheckpoint();
+
+  /**
+   * Restores states of operators.
+  */
```

Comment:
Please expand the documentation for those functions.
I think that the "current checkpoint" is a bit ambiguous. It should specify the prerequisites, that is the `GetCurrentCheckpoint` can be used only after the last output of this epoch was consumed. 

For `RestoreStateFromCheckpoint` it should mention, that we can only resume when the executor was not running. 
Some mention of the post-conditions/state after calling those functions would be welcomed.